backend/gddp/src/main/scala/Gddp.scala

```diff
@@ -93,6 +95,14 @@ object Gddp {
     val xWidth = xSliceStop - xSliceStart + 1
     val yWidth = ySliceStop - ySliceStart + 1
 
+    val pattern = new Regex("2[0-9][0-9][0-9]")
```

Comment:
I am sure that this will work, but for safety's sake it might make sense to add some additional constraints before and after the suspected year to ensure that you don't match the wrong thing.  That will probably never happen with files taken from S3 because of the filenames used, but could conceivably happen from a locally-stored file.